### PR TITLE
fix: Adding xform expression to work around some issues in the docker…

### PR DESCRIPTION
**Pull request by mladkau**
### What this does

Adding an xform expression to work around an issue in the snyk-docker-plugin.
See: https://github.com/snyk/snyk-docker-plugin/issues/218

The xform argument executes a sed like replace expression on the file paths entries in the tar file.

So if a `tar tf image.tar` would return before something like:
./foo/bar

with the xform expression we return now:
foo/bar

### CODE REVIEW

**Review by hisenb3rg — APPROVED** [review]
Can we please add more context to the commit, .e.g. what exactly is transformation doing, and why? 🙏

**Comment by mladkau** [comment]
Re: `What exactly is transformation doing:`

It is taken from this answer: https://serverfault.com/questions/457480/tar-leading-period-slash

The xform argument is a sed-like replace expression of the form: `s/regexp/replace/[flags]`

We are using here: `s:\\./\\?::`

1. We replace the `/` character with `:` 
See the [docs](https://www.gnu.org/software/sed/manual/html_node/The-_0022s_0022-Command.html): 
```
The / characters may be uniformly replaced by any other single character within any given s command. The / character (or whatever other character is used in its stead) can appear in the regexp or replacement only if it is preceded by a \ character. 
```

2. We look for `./` - we need to escape `.` with `\.` and need to escape the escape `\` with `\\` :smile: 

3. We state that this might or might not be there with a `?` which needs to be escaped with `\` which in turn needs to be escaped with `\\`.

4. The last `::` means that we replace it with nothing.

QED :krotik-yay:

re: `Why are we doing this`

We have to work around https://github.com/snyk/snyk-docker-plugin/issues/218 (see also the slack discussion linked in a comment). Which is caused by not adding these [lines](https://github.com/snyk/snyk-docker-plugin/pull/210) from the now closed PR.

TLDR: We're adding here an ugly regular replace expression to work around some fragile code in the snyk-docker-plugin.
